### PR TITLE
fix behavior test with --test-evented-io on windows

### LINKED ISSUE #3717: pub const io_mode = .evented; does not compile when targeting windows
Example program:

```zig
const std = @import("std");
// comment out this line to successfully compile
pub const io_mode = std.io.Mode.evented;
pub fn main() void {
  std.debug.warn("Hello\n");
}
```

Example error when running `zig.exe build-exe events.zig`

```
broken LLVM module found: Instruction does not dominate all uses!
  %743 = load %std.pdb.MsfStream**, %std.pdb.MsfStream*** %92, align 8, !dbg !7527
  %501 = load %std.pdb.MsfStream*, %std.pdb.MsfStream** %743, align 8, !dbg !7549
Instruction does not dominate all uses!
  %743 = load %std.pdb.MsfStream**, %std.pdb.MsfStream*** %92, align 8, !dbg !7527
  %512 = load %std.pdb.MsfStream*, %std.pdb.MsfStream** %743, align 8, !dbg !7555
```

daurnimator indicates this is likely because OutputStream is not yet implemented as async and is required for windows format output

### PR DESCRIPTION
**Pull request by FireFox317**
async io for stdin/out/err is not possible on windows using `CreateIoCompletionPort`. For now always make stdin/out/err blocking.

**Simple file async io works:**
```zig
const std = @import("std");

pub const io_mode = .evented;

pub fn main() !void {
    const test_file = try std.fs.cwd().openFile("test.txt", .{});
    var buf: [256]u8 = undefined;
    const amt = try test_file.read(buf[0..]);
    std.debug.warn("read: {}\n", .{buf[0..amt]});
}
```
Fixed #3717

Edit: See this link why async io on stdin/out/err is not possible on windows: https://tim.mcnamara.nz/post/176613307022/iocp-and-stdio

### CODE REVIEW

**Comment by BarabasGitHub** [comment]
[This answer on stack overflow](https://stackoverflow.com/a/4552494) suggests it's possible to do async io on stdin using [WaitForSingleObject[(https://docs.microsoft.com/en-us/windows/win32/api/synchapi/nf-synchapi-waitforsingleobject) and [ReadConsoleInput](https://docs.microsoft.com/en-us/windows/console/readconsoleinput). 

Apparently they fixed it in tokio, see: https://crates.io/crates/tokio-stdin
And I don't know any Rust, but [async::stdin](https://docs.rs/async-std/1.5.0/async_std/io/struct.Stdin.html) and [tokio::stdin](https://docs.rs/tokio/0.2.13/tokio/io/struct.Stdin.html) seem to suggest you can work around it somehow. Probably not with IO completion ports.

File: lib/std/debug.zig [hunk 1]
```diff
@@ -666,158 +666,160 @@ pub fn openSelfDebugInfo(allocator: *mem.Allocator) anyerror!DebugInfo {
 
 /// TODO resources https://github.com/ziglang/zig/issues/4353
 fn openCoffDebugInfo(allocator: *mem.Allocator, coff_file_path: [:0]const u16) !ModuleDebugInfo {
-    const coff_file = try std.fs.openFileAbsoluteW(coff_file_path.ptr, .{});
-    errdefer coff_file.close();
+    noasync {
+        const coff_file = try std.fs.openFileAbsoluteW(coff_file_path.ptr, .{});
```
Comment (andrewrk) [inline lib/std/debug.zig]: In order to use `noasync` here, the program must guarantee that `openFileAbsoluteW` will not block. What makes this guaranteed? I think `openFileAbsoluteW` is missing an option in the `OpenFlags` argument which would provide this guarantee.

File: lib/std/os/windows.zig [hunk 2]
```diff
@@ -135,6 +135,7 @@ pub fn OpenFileW(
         .SecurityQualityOfService = null,
     };
     var io: IO_STATUS_BLOCK = undefined;
+    const create_options = if (std.io.is_async) FILE_NON_DIRECTORY_FILE else FILE_NON_DIRECTORY_FILE | FILE_SYNCHRONOUS_IO_NONALERT;
```
Comment (andrewrk) [inline lib/std/os/windows.zig]: This should actually be a parameter to OpenFileW I think. Have a look at `std.fs.Dir.openFileW` and the corresponding posix function `openFileZ`, noting the use of `async_block_allowed` field.

File: lib/std/debug.zig (hunk 1 above)
Comment (FireFox317) [reply lib/std/debug.zig]: `noasync` asserts that it won't `suspend` right? On Windows opening a file is always blocking and when opening the file one has to specify whether or not next usages (like write, reads) are async or not, using the `create_options` flags.

**Comment by FireFox317** [comment]
Now stack trace printing also works with `.evented` :)

**Comment by andrewrk** [comment]
=> #5249

**Pull request closed without merging**